app/tests/unit/pages/projects.test.tsx

```diff
@@ -3,56 +3,85 @@ import { Projects, ProjectsQuery } from "../../../pages/cif/projects";
 import { projectsQuery } from "__generated__/projectsQuery.graphql";
+import { MockResolvers } from "relay-test-utils/lib/RelayMockPayloadGenerator";
 
-const environment = createMockEnvironment();
+let environment: RelayMockEnvironment;
+let initialQueryRef;
 
-environment.mock.queueOperationResolver((operation) => {
-  return MockPayloadGenerator.generate(operation, {
-    projectsQuery() {
-      return {
-        query: {
-          session: null,
-          allProjects: {
-            edges: [],
-          },
-          allFormChanges: {
-            edges: [],
-          },
-        },
-      };
-    },
-  });
-});
-
-const query = ProjectsQuery; // can be the same, or just identical
-const variables = {
-  // ACTUAL variables for the invocation goes here
+const defaultMockResolver = {
+  Query() {
+    return {
+      session: { cifUserBySub: {} },
+      allProjects: {
+        edges: [],
+      },
+      pendingNewProjectRevision: null,
+    };
+  },
 };
-environment.mock.queuePendingOperation(query, variables);
 
-describe("The projects page", () => {
-  const initialQueryRef = loadQuery<projectsQuery>(
+const loadProjectsQuery = (
+  mockResolver: MockResolvers = defaultMockResolver
+) => {
+  environment.mock.queueOperationResolver((operation) => {
+    return MockPayloadGenerator.generate(operation, mockResolver);
+  });
+
+  const variables = {};
+  environment.mock.queuePendingOperation(ProjectsQuery, variables);
+  initialQueryRef = loadQuery<projectsQuery>(
     environment,
     ProjectsQuery,
-    {}
+    variables
   );
+};
 
-  it("loads the Create Project Button", async () => {
-    render(
-      <RelayEnvironmentProvider environment={environment}>
-        <Projects CSN={true} preloadedQuery={initialQueryRef} />
-      </RelayEnvironmentProvider>
-    );
+const renderProjects = () =>
+  render(
+    <RelayEnvironmentProvider environment={environment}>
+      <Projects CSN={true} preloadedQuery={initialQueryRef} />
+    </RelayEnvironmentProvider>
+  );
 
-    expect(screen.getAllByRole("button")[1]).toHaveTextContent(
-      "Create Project"
-    );
+describe("The projects page", () => {
+  beforeEach(() => {
+    environment = createMockEnvironment();
   });
 
-  it("calls the Create Project mutation when the Create Project Button is clicked", async () => {
+  it("loads the Add a Project Button", () => {
+    loadProjectsQuery();
+    renderProjects();
+
+    expect(screen.getByText(/Add a Project/i)).toBeInTheDocument();
+  });
+
+  it("renders the Resume Project Draft button if a draft exists", () => {
+    loadProjectsQuery({
+      Query() {
+        return {
+          session: { cifUserBySub: {} },
+          allProjects: {
+            edges: [],
+          },
+          pendingNewProjectRevision: {
+            id: "123",
+          },
+        };
+      },
+    });
+    renderProjects();
+
+    expect(screen.getByText(/resume project draft/i)).toBeInTheDocument();
```

Comment:
We probably want to make sure that the "Add a Project" text is not on the page, since we're not looking at snapshots